src/App.js

```diff
@@ -0,0 +1,101 @@
+import { useEffect, useState } from 'react'
+import { Routes, Route, createSearchParams, useSearchParams, useNavigate } from "react-router-dom"
+import { useDispatch, useSelector } from 'react-redux'
+import 'reactjs-popup/dist/index.css'
+import { fetchMovies } from './data/moviesSlice'
+import { ENDPOINT_SEARCH, ENDPOINT_DISCOVER, ENDPOINT, API_KEY } from './constants'
+import Header from './components/Header'
+import Movies from './components/Movies'
+import Starred from './components/Starred'
+import WatchLater from './components/WatchLater'
+import YouTubePlayer from './components/YoutubePlayer'
+import './app.scss'
+
+const App = () => {
```

Comment:
It would be beneficial to create a dedicated home page component for managing the movie list, which will help to prevent unnecessary re-renders and maintain a focused rendering scope.

Transferring local state and its logic from App.js to child components to minimize re-rendering in the root component. This approach improves performance by reducing the re-render frequency of all components.
 
App.js, as the root component, is responsible for rendering the entire application. Utilizing state in this component might lead to performance bottlenecks due to frequent re-renders across the component tree upon state changes. By delegating stateful responsibilities to dedicated components, re-renders are confined to specific components, optimizing application responsiveness and speed.